src/GAP/gap_to_oscar.jl

```diff
@@ -38,8 +38,66 @@ fmpz_mat(obj::GAP.GAP.GapObj) = gap_to_julia(fmpz_mat, obj)
 ## matrix of GAP rationals or integers to `fmpq_mat`
 function GAP.gap_to_julia(::Type{fmpq_mat}, obj::GAP.GapObj)
     m = GAP.gap_to_julia(Matrix{fmpq}, obj)
-#   return fmpq_mat(size(m, 1), size(m, 2), m) # corrupted object!
     return matrix(FlintQQ, m)
 end
 
 fmpq_mat(obj::GAP.GAP.GapObj) = gap_to_julia(fmpq_mat, obj)
+
+## cache default bases of GAP's cyclotomic fields
+const default_bases_GAP_cyclotomic_fields = AbstractAlgebra.WeakValueDict{Int64, GAP.GapObj}()
+
+function default_basis_GAP_cyclotomic_field(N::Int64, F::GAP.GapObj = GAP.Globals.CyclotomicField(N))
+    get!(default_bases_GAP_cyclotomic_fields, N) do
+      root = GAP.Globals.E(N)
+      elms = [root^i for i in 0:(euler_phi(N)-1)]
+      return GAP.Globals.Basis(F, GAP.GapObj(elms))
+    end
+end
+
+## single GAP cyclotomic to element of cyclotomic field
+function (F::AnticNumberField)(obj::GAP.GapObj)
+    Nemo.iscyclo_type(F) || throw(ArgumentError("F is not a cyclotomic field"))
+    GAP.Globals.IsCyclotomic(obj) || throw(ArgumentError("input is not a GAP cyclotomic"))
+    N = Oscar.get_special(F, :cyclo)
+    mod(N, GAP.Globals.Conductor(obj)) == 0 || throw(ArgumentError("obj does not embed into F"))
+
+    B = default_basis_GAP_cyclotomic_field(N)
+    return F(Vector{fmpz}(GAP.Globals.Coefficients(B, obj)))
+end
+
+## nonempty list of GAP matrices over the same cyclotomic field
+function matrices_over_cyclotomic_field(gapmats::GAP.GapObj)
+    GAP.Globals.IsList(gapmats) || throw(ArgumentError("gapmats is not a GAP list"))
+    GAP.Globals.IsEmpty(gapmats) && throw(ArgumentError("gapmats is empty"))
+    GAP.Globals.ForAll(gapmats, GAP.Globals.IsCyclotomicCollColl) ||
+      throw(ArgumentError("gapmats is not a GAP list of matrices of cyclotomics"))
+
+    gapF = GAP.Globals.FieldOfMatrixList(gapmats)
+    N = GAP.Globals.Conductor(gapF)
+    B = default_basis_GAP_cyclotomic_field(N, gapF)
+    F, z = CyclotomicField(N)
+
+    result = dense_matrix_type(F)[]
+    for mat in gapmats
+      m = GAP.Globals.NumberRows(mat)
+      n = GAP.Globals.NumberColumns(mat)
+      entries = [F(Vector{fmpz}(GAP.Globals.Coefficients(B, mat[i, j])))
+                 for i in 1:m, j in 1:n]
+      push!(result, matrix(F, entries))
+    end
+
+    return result, F, z
+end
+
+## single GAP matrix of cyclotomics
+function matrix(F::AnticNumberField, mat::GAP.GapObj)
+    Nemo.iscyclo_type(F) || throw(ArgumentError("F is not a cyclotomic field"))
+    (GAP.Globals.IsCyclotomicCollColl(mat) && GAP.Globals.IsMatrixOrMatrixObj(mat)) || throw(ArgumentError("mat is not a GAP matrix of cyclotomics"))
+    N = Oscar.get_special(F, :cyclo)
+    B = default_basis_GAP_cyclotomic_field(N)
+    m = GAP.Globals.NumberRows(mat)
+    n = GAP.Globals.NumberColumns(mat)
+    entries = [F(Vector{fmpz}(GAP.Globals.Coefficients(B, mat[i, j])))
+               for i in 1:m, j in 1:n]
```

Comment:
I guess this will produce an error if `F` is too small to represent all entries of `mat`. Perhaps you could add a test case for that situation, just to make sure it really produces an error and not garbage.

Reply:
There is already such a test, the attempt to convert a matrix containing `E(5)` to a matrix over `CyclotomicField(7)`.
(`GAP.Globals.Coefficients` returns `fail` in this case.)